packages/workflow/src/internals.ts

```diff
@@ -964,6 +970,9 @@ export class Activator implements ActivationHandler {
         },
         true
       );
+    } else {
+      // Only keep the first error that bubbles up; subsequent errors will be ignored.
+      if (this.workflowTaskError === undefined) this.workflowTaskError = error;
```

Comment:
Wouldn't it be better to throw a special error type and handle in the unhandled rejection path?

Reply:
I considered that avenue, but personally can't see any strong point in favor of it. And without a strong argument, I'd prefer just avoiding perusing unhandled rejections, because they make it harder for new SDK devs to figure out what's happening.

Reply:
Yeah, my reasoning behind it is that there's now two paths where workflow task errors may be surfaced, making the code harder to follow. The fact that we throw on `tryUnblockConditions` isn't very intuitive to me.